New package: IdealGasThermo v1.0.0

JuliaRegistrator opened the pull request:
- Registering package: IdealGasThermo
- Repository: https://github.com/MIT-LAE/IdealGasThermo.jl
- Created by: @askprash
- Version: v1.0.0
- Commit: 28d76b40bbdfca545f0e4c33efead65f94db14b5
- Reviewed by: @askprash
- Reference: https://github.com/MIT-LAE/IdealGasThermo.jl/commit/28d76b40bbdfca545f0e4c33efead65f94db14b5#commitcomment-155671829
- Description: Julia package to model thermodynamics of ideal gas mixtures.
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd687ee56ae18ce21ce070a67859d4e083b52d584432da0eaa6b0a64efe42e9febcc0da032e9f74d2b36cf1dc59608d839f1a8f6f2b39621fb5a7643c322c55da07fb9c2c7e65ea94e0a806c68e545efc86423544335fd0117ac6f299ba467c05d9d3e2b6c3befda0401bf3b7626e627abfc7462ef34f6345000a2a8871aa59f7fa1bf1a601cc26a93c0032f0daa91cf02b971907819b2acbb5391aecfa91db08334d36000d3e397db0c9bc28c9a33e037b -->

goerz commented:
In https://github.com/JuliaRegistries/General/pull/129139#issuecomment-2808274892 @vinodjanardhanan wrote: 

> IdealGas.jl does the same thing. It also calculates the thermodynamic properties using NASA polynomials

Is there really a need for a separate registered package, then? Is there any chance to for  @askprash collaborate or co-maintain on the existing [IdealGas.jl](https://github.com/vinodjanardhanan/IdealGas.jl)? From a community perspective, having two packages with basically interchangeable names and the same functionality does not seem desirable.

askprash commented:
Thanks for raising this @goerz! My perspective is that there these two packages do some things that are the same but are really meant for different purposes. From what I can tell [`IdealGas.jl`](https://github.com/vinodjanardhanan/IdealGas.jl) is not really _meant_ to be used as a stand alone package (surely it can be by someone who wants to) but is actually a small component of what seems to be a larger reaction kinetics package. I see that @vinodjanardhanan has several other packages that all together comprise a reaction kinetics "meta-package". 

This package here does indeed have the NASA poly calcs but additionally does several other things for example (1) building reduced order mixture representations that are much faster (benchmarked in our docs, but can only be used when certain assumptions are met e.g., in typical ground based or commercial aircraft gas turbines) (2) gas turbine related calculations like compressing gasses, setting flow Mach numbers etc. More importantly, we need these additional functionalities as we work on speeding up [`TASOPT.jl`](https://github.com/MIT-LAE/TASOPT.jl) -- a larger aircraft design package that I just submitted for registration [here](https://github.com/JuliaRegistries/General/pull/129203#issuecomment-2809839786) which will eventually need features in this package that are beyond what `IdealGas.jl` seems to be written for. Additionally, these features are useful beyond our aircraft design code and are going to be the needed in another package we are developing. As far as I can tell a registered package can only depend on other registered packages so not registering this is not an option we'd really like. 

Finally, I think the docs for each of these packaged does a pretty good job of disambiguating use cases for the community if that's the main concern - also open to other name suggestions if that's the main blocker. Let me know what you think!

askprash commented:
Actually looking at this further, @vinodjanardhanan wouldn't it make more sense for all these packages to be modules and not their own packages?
- [StirredReactor.jl](https://github.com/vinodjanardhanan/StirredReactor.jl)
- [BatchReactor.jl](https://github.com/vinodjanardhanan/BatchReactor.jl)
- [PlugFlowReactor.jl](https://github.com/vinodjanardhanan/PlugFlowReactor.jl)
- etc..

These all seem to be single julia files stored as separate packages. It might make sense for them to be grouped together as a single package with each of those being modules, no?

askprash commented:
Hi @goerz could you please help override the auto-merge? Thanks!

goerz commented:
@askprash Done, but you still need to deal with 

>  [compat] entry that is upper-bounded and only includes a finite number of breaking releases: StructArrays

JuliaRegistrator committed:
New package: IdealGasThermo v1.0.0

UUID: d4fb33c0-2494-4384-b7a5-e5d0487b6649
Repo: https://github.com/MIT-LAE/IdealGasThermo.jl.git
Tree: 7b56f71c09206b87684cea6036973dfd877b4e36

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a
